dashboard/dashboardPane.ts

```diff
@@ -0,0 +1,69 @@
+import { PaneDefinition, SolidSession } from "../types"
+import solidUi, { SolidUi } from "solid-ui"
+import paneRegistry from "pane-registry"
+import { NamedNode, sym } from "rdflib"
+import { generateHomepage } from "./homepage"
+
+let panes: any
+let UI: SolidUi
+
+const nodeMode = (typeof module !== "undefined")
+
+if (nodeMode) {
+  UI = solidUi
+  panes = paneRegistry
+} else { // Add to existing mashlib
+  panes = (window as any).panes
+  UI = panes.UI
+}
+
+export const dashboardPane: PaneDefinition = {
+  icon: UI.icons.iconBase + "noun_547570.svg",
+  name: "dashboard",
+  label: (subject) => {
+    if (subject.uri === subject.site().uri) {
+      return "Dashboard"
+    }
+    return null
+  },
+  render: (subject, dom) => {
+    const container = dom.createElement("div")
+    const webId = UI.authn.currentUser()
+    buildPage(container, webId, dom, subject)
+    UI.authn.solidAuthClient.trackSession(async (session: SolidSession) => {
+      container.innerHTML = ""
+      buildPage(container, session ? sym(session.webId) : null, dom, subject)
+    })
+
+    return container
+  }
+}
+
+function buildPage (container: HTMLElement, webId: NamedNode | null, dom: HTMLDocument, subject: NamedNode) {
+  if (!webId) {
+    return buildHomePage(container, subject)
+  }
+  if (webId.site().uri === subject.site().uri) {
+    return buildDashboard(container, dom)
+  }
+  return buildHomePage(container, subject)
+}
+
+function buildDashboard (container: HTMLElement, dom: HTMLDocument) {
+  const outliner = panes.getOutliner(dom)
+  outliner.showDashboard(container)
+}
+
+function buildHomePage (container: HTMLElement, subject: NamedNode) {
+  const wrapper = document.createElement('div')
+  container.appendChild(wrapper)
+  const shadow = wrapper.attachShadow({ mode: 'open' })
```

Comment:
So the homepage is shown inside the data browser in the area reserved for the Pane, is that correct? I also just realised that Shadow DOM isn't supported in IE11, is that something we're OK with?

Reply:
The data browser is currently not working with IE at all - so in my mind, it is not something we can prioritize at the moment.

But we should be mindful of this in the future, so glad that you mentioned this.